server/src/main/java/io/spine/server/integration/ThirdPartyContext.java

```diff
@@ -0,0 +1,190 @@
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.integration;
+
+import com.google.protobuf.Any;
+import io.spine.base.EventMessage;
+import io.spine.core.ActorContext;
+import io.spine.core.Event;
+import io.spine.core.EventContext;
+import io.spine.core.EventId;
+import io.spine.core.UserId;
+import io.spine.server.BoundedContext;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.spine.base.Identifier.newUuid;
+import static io.spine.base.Time.currentTime;
+import static io.spine.protobuf.AnyPacker.pack;
+import static io.spine.server.BoundedContextBuilder.notStoringEvents;
+
+/**
+ * An external non-Spine based upstream system.
+ *
+ * <p>{@code ThirdPartyContext} helps to represent an external system as a Bounded Context. Events
+ * which occur in the external system are converted into domain events of the user's
+ * Bounded Contexts and dispatched via {@link IntegrationBus}.
+ *
+ * @implSpec Note that a {@code ThirdPartyContext} sends a request for external messages to
+ *         other contexts. The {@code ThirdPartyContext} never consumes external messages itself,
+ *         but requires the other Bounded Contexts to send their requests, so that the publishing
+ *         channels are open. Depending of the implementation of
+ *         {@link io.spine.server.transport.TransportFactory transport}, creating
+ *         a {@code ThirdPartyContext} may be an expensive operation. Thus, it is recommended that
+ *         the instances of this class are reused and {@linkplain #close() closed} when they are
+ *         no longer needed.
+ */
+public final class ThirdPartyContext implements AutoCloseable {
+
+    private final BoundedContext context;
+    private final Any producerId;
+
+    /**
+     * Creates a new single-tenant instance of {@code ThirdPartyContext} with the given name.
+     *
+     * @param name
+     *         name of the third-party system
+     */
+    public static ThirdPartyContext singleTenant(String name) {
+        checkNotNull(name);
+        return newContext(name, false);
+    }
+
+    /**
+     * Creates a new multitenant instance of {@code ThirdPartyContext} with the given name.
+     *
+     * @param name
+     *         name of the third-party system
+     */
+    public static ThirdPartyContext multitenant(String name) {
+        checkNotNull(name);
+        return newContext(name, true);
+    }
+
+    private static ThirdPartyContext newContext(String name, boolean multitenant) {
+        BoundedContext context = notStoringEvents(name, multitenant).build();
+        context.integrationBus()
+               .notifyOfCurrentNeeds();
```

Comment:
We need to name this method so that it tells about the fact that Contexts need to speak to each other, maybe again, because a new context appeared in the whole picture of the system.